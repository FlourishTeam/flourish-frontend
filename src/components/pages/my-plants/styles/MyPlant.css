.MyPlant {
    display: flex;
    align-items: center;
    border: 1px inset #ccc;
    width: 100vw;

    .plantImg {
        height: 90px;
    }

    .caption {
        text-decoration: none;
        margin-left: 20px;
    }

    .scientificName {
        font-style: italic;
    }
}