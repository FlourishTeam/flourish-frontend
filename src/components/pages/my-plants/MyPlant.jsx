import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles/MyPlant.css';
// import RemoveFromCollection from '../../reusable/collection/RemoveFromCollection';

const MyPlant = ({  commonName, scientificName, image }) => (

  <figure className={styles.MyPlant}>
    <img
      src={image}
      alt={commonName}
      className={styles.plantImg}
      
    />

    <figcaption className={styles.caption}>
      <p className={styles.commonName}>{commonName}</p>
      <p className={styles.scientificName}>{scientificName}</p>
    </figcaption>
   
  </figure>
);

MyPlant.propTypes = {
  commonName: PropTypes.string.isRequired,
  scientificName: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
};

export default MyPlant;
