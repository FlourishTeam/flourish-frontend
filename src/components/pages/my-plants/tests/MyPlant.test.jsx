import React from 'react';
import { render, cleanup } from '@testing-library/react';
import MyPlant from './MyPlant';

describe('MyPlant component', () => {
  afterEach(() => cleanup());
  it('renders MyPlant', () => {
    const { asFragment } = render(<MyPlant />);
    expect(asFragment()).toMatchSnapshot();
  });
});
