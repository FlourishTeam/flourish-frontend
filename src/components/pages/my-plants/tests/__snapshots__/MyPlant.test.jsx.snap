// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPlant component renders MyPlant 1`] = `
<DocumentFragment>
  <figure
    class="MyPlant"
  >
    <img
      class="plantImg"
    />
    <figcaption
      class="caption"
    >
      <p
        class="commonName"
      />
      <p
        class="scientificName"
      />
    </figcaption>
  </figure>
</DocumentFragment>
`;
